include/boost/next_prior.hpp

```diff
@@ -140,6 +138,24 @@ struct prior_impl1< T, Distance, true >
     }
 };
 
+template< typename T, typename Distance, bool IsIterator = is_iterator< T >::value >
+struct prior_advance_impl :
+    public prior_minus_assign_impl< T, Distance >
+{
+};
+
+template< typename T, typename Distance >
+struct prior_advance_impl< T, Distance, true >
+{
+    static T call(T x, Distance n)
+    {
+        // Avoid negating n to sidestep possible integer overflow
+        std::reverse_iterator< T > rx(x);
```

Comment:
I don't want to trade integer overflow for additional functionality. I suppose, we could switch to `boost::reverse_iterator`, but I would prefer to avoid adding a dependency on Boost.Iterator. In practice, I think `std::reverse_iterator` should work in the case you presented, unless it actively tests for the underlying iterator category.